operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -395,6 +415,40 @@ func (r *TenantReconciler) createOrUpdateClusterResources(ctx context.Context, t
 		retErr = err
 	}
 	klog.Infof("Allow network policy for tenant %s %s", tn.Name, npAOpRes)
+	// Persistant volume claim NFS
+	pvc := v1.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{Name: UserPvcName, Namespace: nsName}}
+
+	pvcOpRes, err := ctrl.CreateOrUpdate(ctx, r.Client, &pvc, func() error {
+		r.updateTnPersistentVolumeClaim(&pvc)
+		return ctrl.SetControllerReference(tn, &pvc, r.Scheme)
+	})
+	if err != nil {
+		klog.Errorf("Unable to create or update PVC for tenant %s -> %s", tn.Name, err)
+		retErr = err
+	}
+	klog.Infof("PVC for tenant %s %s", tn.Name, pvcOpRes)
+
+	if pvc.Status.Phase == v1.ClaimBound {
+		pv := v1.PersistentVolume{ObjectMeta: metav1.ObjectMeta{Name: pvc.Spec.VolumeName}}
+		if err := r.Get(ctx, types.NamespacedName{Name: pv.Name}, &pv); err != nil {
+			klog.Errorf("Unable to get PV for tenant %s -> %s", tn.Name, err)
+			retErr = err
+		} else {
+			pvcSecret := v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: NFSSecretName, Namespace: nsName}}
+			pvcSecOpRes, err := ctrl.CreateOrUpdate(ctx, r.Client, &pvcSecret, func() error {
+				r.updateTnPVCSecret(&pvcSecret, fmt.Sprintf("%s.%s.svc.cluster.local", pv.Spec.CSI.VolumeAttributes["server"], RookCephNamespace), pv.Spec.CSI.VolumeAttributes["share"])
+				return ctrl.SetControllerReference(tn, &pvcSecret, r.Scheme)
+			})
+			if err != nil {
+				klog.Errorf("Unable to create or update PVC Secret for tenant %s -> %s", tn.Name, err)
+				retErr = err
+			}
+			klog.Infof("PVC Secret for tenant %s %s", tn.Name, pvcSecOpRes)
+		}
+	} else if pvc.Status.Phase == v1.ClaimPending {
+		klog.Infof("PVC pending for tenant %s", tn.Name)
+	}
```

Comment:
I would personally extract this to a dedicated function.

Reply:
Moved